sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -432,6 +432,17 @@
                 "description": "Managed network settings to be used for the workspace. If not specified, isolation mode Disabled is the default"
             }
         },
+        "provisionNetworkNow": {
```

Comment:
I think you need to make similar changes in workspace_project.json as well.

Reply:
Project WS is not supported for having a managed vnet provisioned since it associated to its Hub Workspace, so this flag won't be able to be used to Project WS.

Reply:
Thanks for capturing this! I was thinking they are different since one uses `Workspace(...)` and the other uses `Project(...)`. The debug trace shows that they use the same classes in the call stacks, so yes, I need to add the flag to `workspace_project.json` as well. It might work by updating the workspace logic to detect for Project WS specifically to suppress this flag to be populated. Since all others did this, I just follow suite.

Added in a new commit. The test of Project WS creation is as follows:

- dedicated test with Python SDK passed for creating Project WS

```
(env) C:/Program Files/Microsoft VS Code $ C:/repo/azure-sdk-for-python/env/Scripts/python.exe "c:/Users/jingli8/OneDrive - Microsoft/git/tool/rest/ws.py"
Creating AML client object with default auth...
Creating Project WS object...
Class Project: This is an experimental class, and may change at any time. Please see https://aka.ms/azuremlexperimental for more information.
Creating WS resource...
azureml.mlflow could not be imported. Please ensure that latest 'azureml-mlflow' has been installed in the current python environment
The deployment request jing01-ws203-project1-7015758 was accepted. ARM deployment URI for reference:
https://portal.azure.com//#blade/HubsExtension/DeploymentDetailsBlade/overview/id/%2Fsubscriptions%2F0caf7ec9-615a-4491-bad8-64ce023324e1%2FresourceGroups%2Fjing01-rg%2Fproviders%2FMicrosoft.Resources%2Fdeployments%2Fjing01-ws203-project1-7015758
Creating AzureML Workspace: (jing01-ws203-project1  ) ..  Done (20s)
Total time : 22s

Created WS: jing01-ws203-project1 in jing01-rg
Project({'kind': 'project', 'print_as_yaml': False, 'discovery_url': 'https://int.api.azureml-test.ms/discovery', 'mlflow_tracking_uri': 'azureml://int.api.azureml-test.ms/mlflow/v1.0/subscriptions/0caf7ec9-615a-4491-bad8-64ce023324e1/resourceGroups/jing01-rg/providers/Microsoft.MachineLearningServices/workspaces/jing01-ws203-project1', 'workspace_id': '3b423021-be29-4bf6-9f59-2ed19d3c1a97', 'feature_store_settings': None, 'name': 'jing01-ws203-project1', 'description': 'jing01-ws203-project1', 'tags': {'createdByToolkit': 'sdk-v2-1.22.0'}, 'properties': {}, 'id': '/subscriptions/0caf7ec9-615a-4491-bad8-64ce023324e1/resourceGroups/jing01-rg/providers/Microsoft.MachineLearningServices/workspaces/jing01-ws203-project1', 'Resource__source_path': '', 'base_path': 'C:\\Users\\jingli8\\OneDrive - Microsoft\\git\\tool\\rest', 'creation_context': None, 'serialize': <msrest.serialization.Serializer object at 0x000001E47A788460>, 'display_name': 'jing01-ws203-project1', 'location': 'centraluseuap', 'resource_group': 'jing01-rg', 'hbi_workspace': False, 'storage_account': '/subscriptions/0caf7ec9-615a-4491-bad8-64ce023324e1/resourceGroups/jing01-rg/providers/Microsoft.Storage/storageAccounts/jing01wsstorage65b613cac', 'container_registry': None, 'key_vault': '/subscriptions/0caf7ec9-615a-4491-bad8-64ce023324e1/resourceGroups/jing01-rg/providers/Microsoft.KeyVault/vaults/jing01wskeyvault5600347f', 'application_insights': None, 'customer_managed_key': None, 'image_build_compute': None, 'public_network_access': 'Enabled', 'identity': <azure.ai.ml.entities._credentials.IdentityConfiguration object at 0x000001E47A700EE0>, 'primary_user_assigned_identity': None, 'managed_network': <azure.ai.ml.entities._workspace.networking.ManagedNetwork object at 0x000001E47A788E80>, 'provision_network_now': None, 'system_datastores_auth_mode': 'accesskey', 'enable_data_isolation': True, 'allow_roleassignment_on_rg': False, 'Workspace__hub_id': '/subscriptions/0caf7ec9-615a-4491-bad8-64ce023324e1/resourceGroups/jing01-rg/providers/Microsoft.MachineLearningServices/workspaces/jing01-ws203-hub', 'serverless_compute': <azure.ai.ml.entities._workspace.serverless_compute.ServerlessComputeSettings object at 0x000001E47A788820>})
```